app/components/post_list/post/footer/footer.tsx

```diff
@@ -0,0 +1,178 @@
+        },
+        following: {
+            color: theme.buttonBg,
+            ...typography('Heading', 75),
+        },
+        followSeparator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.16),
+            height: 16,
+            marginRight: 12,
+            width: 1,
+        },
+    };
+});
+
+const Footer = ({currentUserId, participants, serverUrl, teamId, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const styles = getStyleSheet(theme);
+    const onUnfollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, false);
+    }), []);
+
+    const onFollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, true);
+    }), []);
+
+    let repliesComponent;
+    let followButton;
+    if (thread.replyCount) {
+        repliesComponent = (
+            <>
+                <View style={styles.replyIconContainer}>
+                    <CompassIcon
+                        name='reply-outline'
+                        size={18}
+                        color={changeOpacity(theme.centerChannelColor, 0.64)}
+                    />
+                </View>
+                <FormattedText
+                    style={styles.replies}
+                    testID={`${testID}.reply_count`}
+                    id='threads.replies'
+                    defaultMessage='{count} {count, plural, one {reply} other {replies}}'
+                    values={{
+                        count: thread.replyCount,
+                    }}
+                />
+            </>
+        );
+    }
+    if (thread.isFollowing) {
+        followButton = (
+            <TouchableOpacity
+                onPress={onUnfollow}
+                style={styles.followingButtonContainer}
+                testID={`${testID}.following`}
+            >
+                <FormattedText
+                    id='threads.following'
+                    defaultMessage='Following'
+                    style={styles.following}
+                />
+            </TouchableOpacity>
+        );
+    } else {
+        followButton = (
+            <>
+                <View style={styles.followSeparator}/>
+                <TouchableOpacity
+                    onPress={onFollow}
+                    style={styles.notFollowingButtonContainer}
+                    testID={`${testID}.follow`}
+                >
+                    <FormattedText
+                        id='threads.follow'
+                        defaultMessage='Follow'
+                        style={styles.notFollowing}
+                    />
+                </TouchableOpacity>
+            </>
+        );
+    }
+
+    const participantsList = useMemo(() => {
+        if (participants?.length) {
+            const orderedParticipantsList = [...participants].reverse();
+            return orderedParticipantsList;
+        }
+        return [];
+    }, [participants]);
```

Comment:
Just a thought here.  Is it efficiently memoizing or should we be checking  `participants.length` instead?

Reply:
I believe the order can change while keeping the same amount of participants. But that should be handled in the `UserAvatarStack` component, so I guess checking for the length should be the way to go as Avinash suggest.